test/byContract/IexecPoco/IexecPoco1.test.ts

```diff
@@ -0,0 +1,201 @@
+            await iexecWrapper.depositInIexecAccount(requester, dealPrice);
+            await iexecWrapper.depositInIexecAccount(scheduler, schedulerStake);
+            // Check balances before.
+            expect(await iexecPoco.balanceOf(requester.address)).to.be.equal(dealPrice);
+            expect(await iexecPoco.frozenOf(requester.address)).to.be.equal(0);
+            expect(await iexecPoco.balanceOf(scheduler.address)).to.be.equal(schedulerStake);
+            expect(await iexecPoco.frozenOf(scheduler.address)).to.be.equal(0);
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            const { appOrderHash, datasetOrderHash, workerpoolOrderHash, requestOrderHash } =
+                iexecWrapper.hashOrders(orders);
+            const dealId = getDealId(iexecWrapper.getDomain(), orders.requester, taskIndex);
+            expect(
+                await IexecPocoAccessors__factory.connect(proxyAddress, anyone).computeDealVolume(
+                    ...orders.toArray(),
+                ),
+            ).to.equal(volume);
+
+            expect(await iexecPocoAsRequester.callStatic.matchOrders(...orders.toArray())).to.equal(
+                dealId,
+            );
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray()))
+                .to.emit(iexecPoco, 'OrdersMatched')
+                .withArgs(
+                    dealId,
+                    appOrderHash,
+                    datasetOrderHash,
+                    workerpoolOrderHash,
+                    requestOrderHash,
+                    volume,
+                );
+            // Check balances after.
+            expect(await iexecPoco.balanceOf(proxyAddress)).to.be.equal(dealPrice + schedulerStake);
+            expect(await iexecPoco.balanceOf(requester.address)).to.be.equal(0);
+            expect(await iexecPoco.frozenOf(requester.address)).to.be.equal(dealPrice);
+            expect(await iexecPoco.balanceOf(scheduler.address)).to.be.equal(0);
+            expect(await iexecPoco.frozenOf(scheduler.address)).to.be.equal(schedulerStake);
+            // Check deal
+            const deal = await iexecPoco.viewDeal(dealId);
+            expect(deal.app.pointer).to.equal(appAddress);
+            expect(deal.app.owner).to.equal(appProvider.address);
+            expect(Number(deal.app.price)).to.equal(appPrice);
+            expect(deal.dataset.pointer).to.equal(datasetAddress);
+            expect(deal.dataset.owner).to.equal(datasetProvider.address);
+            expect(Number(deal.dataset.price)).to.equal(datasetPrice);
+            expect(deal.workerpool.pointer).to.equal(workerpoolAddress);
+            expect(deal.workerpool.owner).to.equal(scheduler.address);
+            expect(Number(deal.workerpool.price)).to.equal(workerpoolPrice);
+            expect(Number(deal.trust)).to.equal(1);
+            expect(Number(deal.category)).to.equal(0);
```

Comment:
I suggest to purchase non zero category, tag, callback & bot to read "actual" values instead of a default zero in storage